commands/climber/toggleleftpiston.py

```diff
@@ -0,0 +1,23 @@
+from commands2 import CommandBase
+from wpilib import DoubleSolenoid
+from subsystems.climbers.leftclimbersubsystem import LeftClimber
+
+
+class ToggleLeftPiston(CommandBase):
+    def __init__(self, climber: LeftClimber) -> None:
+        CommandBase.__init__(self)
+        self.setName(__class__.__name__)
+        self.climber = climber
+        self.addRequirements(climber)
+
+    def initialize(self) -> None:
+        if self.climber.leftClimb.piston.get() == DoubleSolenoid.Value.kForward:
+            self.climber.leftClimb.retractPiston()
+        elif self.climber.leftClimb.piston.get() == DoubleSolenoid.Value.kReverse:
+            self.climber.leftClimb.extendPiston()
+        elif self.climber.leftClimb.piston.get() == DoubleSolenoid.Value.kOff:
+            self.climber.leftClimb.extendPiston()
+
+    # pylint: disable-next=no-self-use
+    def isFinished(self) -> bool:
+        return True
```

Comment:
Luke and I just discussed this and decided that it will probably be easier to have toggles on 2 buttons instead of 4 buttons to do each state. We can also add a SmartDashboard display to show the piston state if that is a problem

Reply:
I HIGHLY advise against toggles. We have seen it countless times over the years and across teams that in the heat-of-the-moment it is easy to lose track of what state toggles are in and there is not enough time (especially if our only output is shuffleboard) to examine the laptop screen. It is incredibly more reliable and efficient to have idempotent commands. If the issue is a lack of buttons we will just plug another xbox controller in.